src/site/content/en/blog/intersectionobserver-v2/index.md

```diff
@@ -0,0 +1,240 @@
+    }
+  }
+};
+
+const observer = new IntersectionObserver(onIntersection);
+observer.observe(document.querySelector('#some-target'));
+```
+
+## What's challenging with Intersection Observer&nbsp;v1?
+
+To be clear, Intersection Observer&nbsp;v1 is great, but it's not perfect. There are
+some corner cases where the API falls short. Let's have a closer look!
+The Intersection Observer&nbsp;v1 API can perfectly tell you when an element is scrolled into the
+window's viewport, but it *doesn't* tell you whether the element is covered
+by any other page content (that is, when the element is occluded) or whether
+the element's visual display has been modified by visual effects like `transform`, `opacity`,
+`filter`, etc., which *effectively* can make it invisible.
+
+Now while for an element in the top-level document this information can be determined by analyzing
+the DOM via JavaScript, for example via
+[`DocumentOrShadowRoot.elementFromPoint()`](https://developer.mozilla.org/en-US/docs/Web/API/DocumentOrShadowRoot/elementFromPoint)
+and then digging deeper, the same information cannot be obtained if the element in question is
+located in a third-party iframe.
+
+## Why is actual visibility such a big deal?
+
+The Internet is, unfortunately, a place that attracts bad actors with even worse intentions.
+For example, a shady publisher that serves pay-per-click ads on a content site might be incentivized
+to trick people into clicking their ads to increase the publisher's ad payout (at least
+for a short period, until the ad network catches them).
+Typically, such ads are served in iframes.
+Now if the publisher wanted to get users to click such ads, they could make the ad iframes
+completely transparent by applying a CSS rule `iframe { opacity: 0; }` and overlaying the iframes
+on top of something attractive, like a cute cat video that users would actually want to click.
+This is called *clickjacking*.
+You can see such a clickjacking attack in action in the upper section of this
+[demo](https://trick-ad-click.glitch.me/) (try "watching" the 🐈 cat video
+and ☑️ activate "trick mode").
+You will notice that the ad in the iframe "thinks" it received legitimate clicks, even if it was
+completely transparent when you (pretendedly involuntarily) clicked it.
+
+{% Img src="image/8WbTDNrhLsU0El80frMBGE4eMCD3/oSQhudX6lenD4Pld7PJD.png", alt="Tricking a user into clicking an ad by styling it transparent and overlaying it on top of something attractive.", width="800", height="533" %}
+
+## How does Intersection Observer&nbsp;v2 fix this?
+
+Intersection Observer&nbsp;v2 introduces the concept of tracking the actual "visibility" of a target
+element as a human being would define it.
+By setting an option in the
+[`IntersectionObserver` constructor](https://developer.mozilla.org/en-US/docs/Web/API/IntersectionObserver/IntersectionObserver),
+intersecting
+[`IntersectionObserverEntry`](https://developer.mozilla.org/en-US/docs/Web/API/IntersectionObserverEntry)s
```

Comment:
I don't understand what's after the comma.

Reply:
Yeah, it was weird. Rephrased.